app/components/network/networkController.js

```diff
@@ -19,12 +19,87 @@ function ($scope, $state, $stateParams, Network, Messages) {
     });
   };
 
-  $('#loadingViewSpinner').show();
-  Network.get({id: $stateParams.id}, function (d) {
-    $scope.network = d;
-    $('#loadingViewSpinner').hide();
-  }, function (e) {
-    $('#loadingViewSpinner').hide();
-    Messages.error("Failure", e, "Unable to retrieve network info");
+  $scope.containerLeaveNetwork = function containerLeaveNetwork(network, containerId) {
+    $('#loadingViewSpinner').show();
+    Network.disconnect({id: $stateParams.id}, { Container: containerId, Force: false }, function (d) {
+      if (d.message) {
+        $('#loadingViewSpinner').hide();
+        Messages.send("Error", {}, d.message);
+      } else {
+        $('#loadingViewSpinner').hide();
+        Messages.send("Container left network", $stateParams.id);
+        $state.go('network', {id: network.Id}, {reload: true});
+      }
+    }, function (e) {
+      $('#loadingViewSpinner').hide();
+      Messages.error("Failure", e, "Unable to disconnect container from network");
+    });
+  };
+
+  function getNetwork() {
+    $('#loadingViewSpinner').show();
+    Network.get({id: $stateParams.id}, function (d) {
+      $scope.network = d;
+      getContainersInNetwork(d);
+      $('#loadingViewSpinner').hide();
+    }, function (e) {
+      $('#loadingViewSpinner').hide();
+      Messages.error("Failure", e, "Unable to retrieve network info");
+    });
+  };
+
+  function getContainersInNetwork(network) {
+    if (network.Containers) {
+      Container.query({
+        filters: {network: [$stateParams.id]}
+      }, function (containersInNetwork) {
```

Comment:
The code to filter containers seems complicated, just re-use the same algorithm as in other places for now (till it's implemented on the API side).

E.g. associate the containersInNetwork to the scope variable (you're defining one at the moment, but it's not used in the HTML template:

```javascript
function (containersInNetwork) {
  var containers = containersInNetwork;
  if ($scope.containersToHideLabels) {
    containers = ContainerHelper.hideContainers(containersInNetwork, $scope.containersToHideLabels);
  }
  $scope.containersInNetwork = containers;
  // Remove the rest
}
```

And then iterate on it in the HTML:

```
<tr ng-repeat="network in containersInNetwork">
              <td><a ui-sref="container({id: network.Id})">{{ network.Name }}</a></td>
              <td>{{ network.IPv4Address || '-' }}</td>
              <td>{{ network.IPv6Address || '-' }}</td>
              <td>{{ network.MacAddress || '-' }}</td>
              ...
```

You'll also need to update the `containerLeaveNetwork` function.